egui_extras/src/table.rs

```diff
@@ -352,24 +352,145 @@ pub struct TableBody<'a> {
 }
 
 impl<'a> TableBody<'a> {
+    fn delta(&self) -> f32 {
+        self.start_y - self.layout.current_y()
+    }
```

Comment:
I think this is a weird name and a weird behavior - the _negative_ amount of y progress?

I think this would be better as

```suggestion
    fn y_progress(&self) -> f32 {
        self.layout.current_y() - self.start_y
    }
```

so that the returned value is positive (or am I misunderstanding it?)

Reply:
I could also be misunderstanding, but if you look at the way [delta was calculated prior to this PRd in `TableBody::rows`](https://github.com/emilk/egui/blob/5dff1e42c6ec8a3740c9cf0b18400743d73a0266/egui_extras/src/table.rs#L359-L363) the result was expected to be negative if any buffer spaces was needed above the visible range. My intent with this `delta` helper method was to clean that up a little and to reuse the calculation in `TableBody::heterogeneous_rows`.

So I agree with the intent of your feedback here (ie that it's easier to reason about a positive result), just not with the suggested change since to my understanding what I've written should always result in >= 0 value for delta. But my understanding is definitely limited since I don't have a good conceptual understanding of the difference between `self.start_y` and `self.layout.current_y()` nor even where the origin of the coordinate system in egui is. My guess is that `self.start_y` is the coordinate of the top of the visible rectangle and `self.layout.current_y()` is the top of the "virtual" rectangle pushed upward by the `TableRow`. Or something like that. All I know is that if I flip the subtraction operands around then I have to either negate the result or flip a bunch of comparison operands.

I'll definitely make the suggested name change though.